components/builder-scheduler/src/data_store.rs

```diff
@@ -262,6 +265,70 @@ impl DataStore {
+                                   owner_id bigint,
+                                   job_state text,
+                                   project_name text,
+                                   net_error_code int,
+                                   net_error_msg text,
+                                   build_started_at timestamptz,
+                                   build_finished_at timestamptz,
+                                   created_at timestamptz DEFAULT now()
+                            )"#,
+        )?;
+
+        // Insert a new job status
+        migrator.migrate("scheduler",
+                            r#"CREATE OR REPLACE FUNCTION insert_job_status_v1 (
+                               job_id bigint,
+                               owner_id bigint,
+                               job_state text,
+                               project_name text,
+                               net_error_code int,
+                               net_error_msg text,
+                               build_started_at timestamptz,
+                               build_finished_at timestamptz
+                           ) RETURNS SETOF job_status AS $$
+                                   BEGIN
+                                       RETURN QUERY INSERT INTO job_status (job_id, owner_id, job_state, project_name, net_error_code, net_error_msg, build_started_at, build_finished_at)
+                                           VALUES (job_id, owner_id, job_state, project_name, net_error_code, net_error_msg, build_started_at, build_finished_at)
+                                           RETURNING *;
+                                       RETURN;
+                                   END
+                               $$ LANGUAGE plpgsql VOLATILE
+                               "#)?;
+
+        migrator.migrate("scheduler",
+                        r#"CREATE OR REPLACE FUNCTION get_job_status_v1 (integer) RETURNS SETOF job_status AS $$
+                               BEGIN
+                                   RETURN QUERY SELECT * FROM job_status
+                                        ORDER BY created_at ASC
+                                        LIMIT $1;
+                                   RETURN;
+                               END
+                               $$ LANGUAGE plpgsql VOLATILE"#)?;
+
+        migrator.migrate(
+            "scheduler",
+            r#"CREATE OR REPLACE FUNCTION delete_job_status_v1 (integer) RETURNS void AS $$
+                              BEGIN
+                                  DELETE FROM job_status
+                                  WHERE id IN (
+                                      SELECT id
+                                      FROM job_status
+                                      ORDER BY created_at ASC
+                                      LIMIT $1
+                                  );
+                              END
+                              $$ LANGUAGE plpgsql VOLATILE"#,
```

Comment:
Fair enough. My primary concern is to avoid the temptation of including extra logic in postgres procs and keeping them in rust but that may be a different topic entirely.